### ISSUE TITLE
Kitti result

### ISSUE THREAD

**Issue by DezeZhao** [rayguan97/M3DETR#7]
hello, I am very interested in your work. So I do exp. on KITTI dataset,  but I found the result is lower than PV-RCNN and don't reproduce the same / close results as your paper both on R11 and R40 metrics. Could you please share the checkpoint? Thank you for your reply.

**Comment by DezeZhao** [comment]
<img width="631" alt="image" src="https://user-images.githubusercontent.com/44592986/227881012-701a558d-bad6-4c96-8dc8-c8b4cdb7733d.png">
<img width="533" alt="image" src="https://user-images.githubusercontent.com/44592986/227881105-bc2eeae9-fc7e-4614-ac36-193eb465b27e.png">

**Comment by DezeZhao** [comment]
all above are the best results I evaluated

**Comment by rayguan97** [comment]
Hi,

Thank you for the feedback and interest in our code. We only use KITTI validation set for ablation purpose (Table 5). It might take some time for us to locate the training weights on the KITTI dataset on the validation set. We will try to follow up with you by this weekend and see if we can still locate the original checkpoints on the training set. 

In Table 4, the model is trained on both training and validation sets, and it is evaluated on the testing set. Those numbers can be found on the official KITTI benchmark following [this link](https://www.cvlibs.net/datasets/kitti/eval_object_detail.php?&result=e615ecb9071878f90e6b5814394078159b4f0279). 

In addition, we indeed observe such variations on the KITTI dataset (the performance of moderate drop sightly more than the easy and hard classes, when you modify the training parameters and config settings). It might take some time to tune the model to get the best performance. If possible, we recommend that you try our model on the Waymo dataset, which would produce more stable and consistent results. The checkpoints on Waymo dataset can be found [here](https://github.com/rayguan97/M3DETR/blob/main/MODEL_ZOO.md).

Best,
Tianrui

**Comment by DezeZhao** [comment]
OK, I get it, thank you first. But the result I evaluated above without any training parameters changed on training set and validation set. I totally trained with your code. I trained the model for 3 times for generalization. Unfortunately, I seem to get the same result. The following results are the third time training:
<img width="510" alt="image" src="https://user-images.githubusercontent.com/44592986/228113682-1994fd34-0ce7-4750-9bd5-11ea92333376.png">
<img width="500" alt="image" src="https://user-images.githubusercontent.com/44592986/228113715-80c31f3e-76aa-4752-aba9-e7157a77757c.png">
<img width="691" alt="image" src="https://user-images.githubusercontent.com/44592986/228113849-0c91d311-a038-45d3-8ba6-8147e2736217.png">

**Comment by DezeZhao** [comment]
It seems that the results are close to line 1 of table 5. But I am sure I use the config of Rel. Trans & Rep. and Scal. Trans.  The follow is the config:
![iShot_2023-03-28_10 47 31](https://user-images.githubusercontent.com/44592986/228114874-68ec8900-a07e-44d9-b87e-0d9b098d90c0.jpg)

**Comment by DezeZhao** [comment]
> 

@rayguan97 tianrui, hi! Excuse me, have you found the checkpoint of kitti dataset? Thank you.

**Comment by rayguan97** [comment]
Hi,

Sorry about the delay. I recover the pth file and the config file shown in table 5. Please try [this config](https://github.com/rayguan97/M3DETR/blob/main/tools/cfgs/m3detr_models/m3detr_kitti2.yaml) and [weights](https://drive.google.com/file/d/1Jwr9keDHVabebtf-ApSs7BH8-RDQIUmj/view?usp=sharing) and see if you can reproduce the result on validation set.


Best,
Tianrui

**Comment by DezeZhao** [comment]
ok thank you for your reply! I will try it now.

**Comment by DezeZhao** [comment]
hi tianrui! Could you please show me your training config on KITTI, such as gpu number and batch size?

**Comment by rayguan97** [comment]
I'm not following your question. I attached the weights and the config file to run the training and testing. Those information can be found in the config file.

**Comment by DezeZhao** [comment]
oh yeah. Another question, did you submit the last epoch (e.g. 80th) to the test set server to get the test results? or the best checkpoint on the val set to get the test results?

**Comment by rayguan97** [comment]
We’ve clarified that in our conversation. The model we submit to test server is trained on both trainings and validation set. The current one is trained on the train set and test on validation set.

If you want to submit to the test server, we recommend you train the model on both training and validation set. You are welcome to do so, but you can also find our performance on the official benchmark page.

________________________________

**Comment by rayguan97** [comment]
Closing the issue now. We hope that our answer is helpful.
